### ISSUE TITLE
Suggestion for cancel_order

### ISSUE THREAD

**Issue by yogat3ch** [yogat3ch/AlpacaforR#2]
Hi @jagg19, 
I'm stoked to have just found this package! I've been in the same boat for about a year of debating whether I need to make the switch to Python for this quantitative analysis project in R in order to take it live though I've put so much work into it already! Thank you so much for creating this package! I'm happy to contribute what I can to develop the package and documentation as I use it!

Thus far I have a small suggestion for cancel_order: 

I was just using cancel_order as follows:
```
cancel_order(orders$id[1])
Order ID  for 7976525f-cac3-415e-9269-1273fd25a73d was successfully canceled. 
```
However, when I checked the order with get_orders, and via the browser, the order was not actually canceled, I believe because the ticker was not specified, or because the order id submitted was interpreted as the ticker.
I also noticed that if two orders are placed for the same symbol, cancel_order will error:
```
submit_order(ticker = "AMZN", qty = as.character(1), side = "buy", type = "stop_limit", time_in_force = "day", limit_price = as.character(1896), stop_price = as.character(1896))
submit_order(ticker = "AMZN", qty = as.character(1), side = "buy", type = "stop_limit", time_in_force = "gtc", limit_price = as.character(1896), stop_price = as.character(1896))
cancel_order("AMZN")
# Error in is.url(url) : length(url) == 1 is not TRUE
```

Looking at the code it appears that because the first argument should be the ticker though what was supplied was the id and the function used this as the ticker and accidentally confirmed that the order was canceled though it actually wasn't. 

This function could probably be simplified to accept as it's first argument `ticker_id` as the ticker symbol or id with documentation as such:
```
#' @param ticker_id The ticker symbol or the order id.
```
and to the function could be added the following:
```
if (nchar(ticker_id) > 15) {
  order_id <- ticker_id
  ticker <- open_orders$symbol[open_orders$id == order_id]
  } else {
 ticker <- ticker_id 
 open_orders_sym <- grep(ticker, open_orders$symbol, ignore.case = T)
 # If more than one order is open
 if (length(open_orders_sym > 1)) message(paste0("More than one order open for ",ticker,", most recent order placed at ", lubridate::with_tz(as.POSIXlt(open_orders$submitted_at[open_orders_sym[1]], tz = "UTC", tryFormats = c("%Y-%m-%dT%H:%M:%OS")), Sys.timezone())," will be canceled"))
 order_id <- open_orders[open_orders_sym[1], "id", drop = T]
 }
```
lubridate will need to be added to the depends for the timezone coercion in the message, or a simplified but less informative version is possible:

```
if (nchar(ticker_id) > 15) {
  order_id <- ticker_id
  ticker <- open_orders$symbol[open_orders$id == order_id]
  } else {
 ticker <- ticker_id 
 open_orders_sym <- grep(ticker, open_orders$symbol, ignore.case = T)
 # If more than one order is open
 if (length(open_orders_sym > 1)) message(paste0("More than one order open for ",ticker,", most recent order will be canceled"))
 order_id <- open_orders[open_orders_sym[1], "id", drop = T]
 }
```
This will allow for magrittr piping `%>%` of either ticker symbols or order ids into the function, and makes it such that the ticker symbol is no longer required to be upper case. 

Happy to make these changes if you're busy. 

Thank you again!

**Comment by jagg19** [comment]
Hi @yogat3ch,

Thanks so much for the great feedback and suggestions!! I've been away on other projects and have had this as a priority for a while now. Now that I have some more time, its underway! I really appreciate you for pointing this out, and I am open to more suggestions as you use it! 

When I first created the package/functions, I intended to always use the supplied arguments inside the function. For example, I would do `cancel_order(order_id = orders$id[1])` if I wanted to go the `$id` route. I really like your implementation, and I agree it simplifies this function, so I went ahead and implemented this. 

I agree R is missing an easy way to implement strategies in a live environment, and I think Alpaca has made it easy to develop on their platform to add that support.

Let me know if this is what you had in mind, so we can close this. Thanks!

**Comment by yogat3ch** [comment]
Hi @jagg19,
Certainly, super excited to be integrating this package into the strategy that's underway. It's so straightforward and is simplifying things immensely. 

Indeed, the logic you had intended was clear once I took a look at the actual function, I hoped to preserve and simplify it with the suggestion. I was humming along testing out the package using tidyverse syntax and ran into that exception with the success message but the order still being open and that spurred the idea. 
I definitely appreciate the commenting and indentation you added in your implementation. It's much clearer now! It gave me some ideas of how to provide more straightforward, well-commented code when making suggestions such that it's more legible in a collaborative context.

Thanks!

PS In the submit orders documentation it indicates that arguments that one would think would typically be numeric are to be submitted "as a string". The documentation could probably be simplified by just adding one line of code to the function that coerces all arguments to string regardless of their input type. This should work for dates and numerics alike so the "as a string" bits in the documentation can probably be removed after that:
`bodyl <- lapply(bodyl, as.character)`

Thanks for your prompt response!

**Comment by jagg19** [comment]
Hi @yogat3ch 

Thanks for the feedback on submit_order! I added that line in and simplified the documentation. 

To stay consistent with the case-insensitivity in cancel_orders ticker argument, I added 
'if(grepl("^[[:lower:]]+$",ticker)){ticker <- toupper(ticker)}` to add case-insensitivity in the submit_orders ticker argument as well. I appreciate your help, and thanks for all of the feedback! I'm going to close this issue here. 

Feel free to reach out whenever!
